cartridge-cli.lua

```diff
@@ -680,21 +681,32 @@ end
 
 -- * ---------------- Project-related functions ----------------
 
+local function format_version(major, minor, patch)
+    major = major or 0
+    minor = minor or 0
+    patch = patch or 0
+    return string.format('%s.%s.%s', major, minor, patch)
+end
+
 local function normalize_version(str)
     local patterns = {
+        "(%d+)%.(%d+)%.(%d+)-(%d+)-(%g+)",
         "(%d+)%.(%d+)%.(%d+)",
         "(%d+)%.(%d+)",
         "(%d+)"
     }
 
     for _, pattern in ipairs(patterns) do
-        local major, minor, patch = string.match(str, pattern)
-
+        local major, minor, patch, commit_num, commit_hash = string.match(str, pattern)
         if major ~= nil then
-            minor = minor or '0'
-            patch = patch or '0'
-
-            return {major, minor, patch}
+            local release
+            if commit_num ~= nil then
+                release = tostring(commit_num)
+            end
+            if commit_hash ~= nil then
+                release = string.format('%s-%s', release, commit_hash)
+            end
```

Comment:
Maybe release should be set to 0 by deafault?

Reply:
fixed

Reply:
If user specifies `--version 0.1.0-g8bce594e`, he get `<name>-0.1.0-0-g8bce594e.<type>` package, not  `<name>-0.1.0-g8bce594e.<type>` as he expected.

Maybe it would be better to do smth like this?

```lua

            local release = '0'

            if count ~= nil and hash ~= nil then
                release = string.format('%s-%s', count, hash)
            elseif count ~= nil then
                release = tostring(count)
            elseif hash ~= nil then
                release = tostring(hash)
            end

            return format_version(major, minor, patch), release
```

Reply:
it could be quite strange, but ok, let's support it